lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -118,6 +123,33 @@ static Optional<APSInt> getExtendedConstant(Value operand, Attribute constant,
   return {};
 }
 
+/// Determine the value of a constant operand for the sake of constant folding.
+/// This will map `invalidvalue` to a zero value of the corresopnding type,
+/// which aligns with how the Scala FIRRTL compiler handles invalids.
```

Comment:
```suggestion
/// Determine the value of a constant operand for the sake of constant folding.
/// This will map `invalidvalue` to a zero value of the corresopnding type,
/// which aligns with how the Scala FIRRTL compiler handles invalids in most
/// cases. For a full discussion of this see the FIRRTL Rationale document.
```

Reply:
Awesome thanks, applied this manually in the rebase.